sim/core/apl_values_dot.go

```diff
@@ -194,23 +194,40 @@ func (value *APLValueDotTickFrequency) String() string {
 	return fmt.Sprintf("Dot Tick Frequency(%s)", value.dot.Get().Spell.ActionID)
 }
 
-type APLValueDotPercentIncrease struct {
+type APLValueDotIncreaseCheck struct {
 	DefaultAPLValueImpl
-	spell  *Spell
-	target *Unit
+	spell    *Spell
+	target   *Unit
+	baseName string
 }
 
-func (rot *APLRotation) newValueDotPercentIncrease(config *proto.APLValueDotPercentIncrease, _ *proto.UUID) APLValue {
-	spell := rot.GetAPLSpell(config.SpellId)
+func (rot *APLRotation) newDotIncreaseValue(baseName string, spellId *proto.ActionID, targetUnit *proto.UnitReference) *APLValueDotIncreaseCheck {
+	spell := rot.GetAPLSpell(spellId)
```

Comment:
I would pass in `config *proto.APLValueDotPercentIncrease` to the constructor rather than the separate `spellId` and `targetUnit` fields, since all three child values use the same config struct.